client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCacheManager.java

```diff
@@ -411,6 +432,63 @@ public ChannelFactory getChannelFactory() {
       return channelFactory;
    }
 
+   private TransactionManager getTransactionManager(TransactionManager override) {
+      try {
+         return override == null ? configuration.transaction().transactionManagerLookup()
+               .getTransactionManager() : override;
+      } catch (Exception e) {
+         throw new HotRodClientException(e);
+      }
+   }
+
+   private TransactionMode getTransactionMode(TransactionMode override) {
+      return override == null ? configuration.transaction().transactionMode() : override;
+   }
+
+   private TransactionTable getTransactionTable(TransactionMode transactionMode) {
+      switch (transactionMode) {
+         case NON_XA:
+            return syncTransactionTable;
```

Comment:
`NON_XA` means `useSynchronization(true)`, which hides all commit errors from the user. Couldn't we support at least `NON_DURABLE_XA`?

Reply:
I split the work by JIRA. this is the first one. then I'll add XA and finally recovery